### PR TITLE
optimize (u)int8 vectorized operator*

### PR DESCRIPTION
**Pull request by mikekgfb**
Summary: optimize (u)int8 vectorized operator*

Test Plan: sandcastle github

Differential Revision: D52318192

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by jgong5 — COMMENTED** [review]
LGTM. Can you add avx512 counterpart too?

**Comment by mikekgfb** [comment]
> LGTM. Can you add avx512 counterpart too?

Submitted as #116346

**Review by hl475 — APPROVED** [review]

**Review by malfet — APPROVED** [review]

File: aten/src/ATen/cpu/vec/vec256/vec256_int.h [hunk 1]
```diff
@@ -960,13 +960,37 @@ Vectorized<T> inline int_elementwise_binary_256(const Vectorized<T>& a, const Ve
 template <>
 Vectorized<int8_t> inline operator*(const Vectorized<int8_t>& a, const Vectorized<int8_t>& b) {
   // We don't have an instruction for multiplying int8_t
```
Comment (malfet) [inline aten/src/ATen/cpu/vec/vec256/vec256_int.h:962]:
```suggestion
  // We don't have an instruction for multiplying int8_t, so upcast to int16_t and then mask
```

**Pull request closed without merging**